deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/NeuralNetConfiguration.java

```diff
@@ -261,6 +262,19 @@ public ListBuilder layer(Layer layer){
             return layerwise;
         }
 
+        @Override
+        public ListBuilder setInputType(InputType inputType){
+            return (ListBuilder)super.setInputType(inputType);
+        }
+
```

Comment:
Since my ML-level hasn't reached anything else than feedforward at this point, this might be utterly meaningless: Are you sure it makes sense for a ListBuilder to accept all the different types of InputType? Will the ListBuilder - in this usage scenario - be used in a non-feedforward setting? I had this idea that these other types would be used "within" the network, but not as the top, "receiving" part? As I said, might be totally off base here.